### PR TITLE
Armageddon

### PR DESCRIPTION
**Pull request by Przemog1**

### CODE REVIEW

File: src/nbl/video/CPropertyPoolHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+} inBuff[_IRR_BUILTIN_PROPERTY_COUNT_];
+layout(set=0, binding=2) writeonly restrict buffer OutData
+{
+    uint data[];
+} outBuff[_IRR_BUILTIN_PROPERTY_COUNT_];
+
+
+#if 0 // optimization
+uint shared workgroupShared[_IRR_BUILTIN_PROPERTY_COPY_GROUP_SIZE_];
+#endif
```
Comment (devshgraphicsprogramming) [inline src/nbl/video/CPropertyPoolHandler.cpp:32]: watch out for `IRR` macros

**Comment by devshgraphicsprogramming** [comment]
`src/nbl/asset/CGLSLScanBuiltinIncludeLoader.cpp` can be safely removed from the engine

**Comment by Przemog1** [comment]
> `src/nbl/asset/CGLSLScanBuiltinIncludeLoader.cpp` can be safely removed from the engine

you mean `src/nbl/asset/CGLSLScanBuiltinIncludeLoader.h`?

File: src/nbl/asset/CMeshManipulator.cpp [hunk 2]
```diff
@@ -0,0 +1,1558 @@
...
+{
+namespace asset
+{
+
+//! Flips the direction of surfaces. Changes backfacing triangles to frontfacing
+//! triangles and vice versa.
+//! \param mesh: Mesh on which the operation is performed.
+void IMeshManipulator::flipSurfaces(ICPUMeshBuffer* inbuffer)
+{
+#ifdef OLD_SHADERS
```
Comment (devshgraphicsprogramming) [inline src/nbl/asset/CMeshManipulator.cpp:29]: @Przemog1 or @Crisspl , this is a TODO

File: include/nbl/core/Types.h [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	//#define _CRT_NONSTDC_NO_DEPRECATE 1
+#endif
+
+
+//! creates four CC codes used in Irrlicht for simple ids
+/** some compilers can create those by directly writing the
+code like 'code', but some generate warnings so we use this macro here */
+#define MAKE_NBL_ID(c0, c1, c2, c3) \
+		((uint32_t)(uint8_t)(c0) | ((uint32_t)(uint8_t)(c1) << 8) | \
+		((uint32_t)(uint8_t)(c2) << 16) | ((uint32_t)(uint8_t)(c3) << 24 ))
```
Comment (devshgraphicsprogramming) [inline include/nbl/core/Types.h:141]: throw this out

File: include/nbl/core/Types.h [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+#endif
+
+
+//! ignore VC8 warning deprecated
+/** The microsoft compiler */
+#if defined(_NBL_WINDOWS_API_) && defined(_MSC_VER) && (_MSC_VER >= 1400)
+	//#pragma warning( disable: 4996)
+	//#define _CRT_SECURE_NO_DEPRECATE 1
+	//#define _CRT_NONSTDC_NO_DEPRECATE 1
+#endif
```
Comment (devshgraphicsprogramming) [inline include/nbl/core/Types.h:133]: throw this out

File: include/nbl/core/Types.h [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+#elif !defined(__CYGWIN__)
+#define swprintf _snwprintf
+#define snprintf _snprintf
+#endif
+
+#endif // _NBL_WINDOWS_API_
+
+
+
+#define _NBL_TEXT(X) X
```
Comment (devshgraphicsprogramming) [inline include/nbl/core/Types.h:109]: throw this out

File: include/nbl/core/Types.h [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+//! and int swprintf(wchar_t *wcs, size_t maxlen, const wchar_t *format, ...);
+#if defined(_MSC_VER) && _MSC_VER > 1310 && !defined (_WIN32_WCE)
+#define swprintf swprintf_s
+#define snprintf sprintf_s
+#elif !defined(__CYGWIN__)
+#define swprintf _snwprintf
+#define snprintf _snprintf
+#endif
+
+#endif // _NBL_WINDOWS_API_
```
Comment (devshgraphicsprogramming) [inline include/nbl/core/Types.h:105]: @AnastaZIuk help us to throw this out (you know the C++11 equivalents now)

File: include/nbl/core/Types.h [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+//! creates four CC codes used in Irrlicht for simple ids
+/** some compilers can create those by directly writing the
+code like 'code', but some generate warnings so we use this macro here */
+#define MAKE_NBL_ID(c0, c1, c2, c3) \
+		((uint32_t)(uint8_t)(c0) | ((uint32_t)(uint8_t)(c1) << 8) | \
+		((uint32_t)(uint8_t)(c2) << 16) | ((uint32_t)(uint8_t)(c3) << 24 ))
+
+#if defined(__BORLANDC__) || defined (__BCPLUSPLUS__)
+#define _strcmpi(a,b) strcmpi(a,b)
+#endif
```
Comment (devshgraphicsprogramming) [inline include/nbl/core/Types.h:145]: borland as a compiler is dead

File: source/Nabla/COpenGLDriver.cpp [hunk 8]
```diff
@@ -0,0 +1,3266 @@
...
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+#ifdef _NBL_LINUX_X11_RANDR_
+#include <X11/extensions/Xrandr.h>
+#endif
+#endif
+
+namespace
+{
+class AMDbugfixCompiler : public spirv_cross::Compiler
```
Comment (devshgraphicsprogramming) [inline source/Nabla/COpenGLDriver.cpp:50]: what is this still doing here, I thought it was removed?

File: source/Nabla/os.h [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+			static inline uint16_t byteswap(uint16_t num) { return bswap_16(num); }
+			static inline int16_t byteswap(int16_t num) { return bswap_16(num); }
+			static inline uint32_t byteswap(uint32_t num) { return bswap_32(num); }
+			static inline int32_t byteswap(int32_t num) { return bswap_32(num); }
+			static inline float byteswap(float num) { uint32_t tmp = core::IR(num); tmp = bswap_32(tmp); return core::FR(tmp); }
+			// prevent accidental byte swapping of chars
+			static inline uint8_t  byteswap(uint8_t num) { return num; }
+			static inline int8_t  byteswap(int8_t num) { return num; }
+	};
```
Comment (devshgraphicsprogramming) [inline source/Nabla/os.h:57]: do we even use this crap?

**Comment by devshgraphicsprogramming** [comment]
@Przemog1 @Crisspl @Hazardu @AnastaZIuk address the comments in separate PRs

**Pull request merged**